vertical-pod-autoscaler/pkg/updater/priority/scaling_direction_pod_eviction_admission.go

```diff
@@ -0,0 +1,104 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package priority
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	vpa_types "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	vpa_utils "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/vpa"
+)
+
+// NewScalingDirectionPodEvictionAdmission creates a PodEvictionAdmission object.
+// It admits Pods for eviction based on the scaling direction, i.e. if a resource is scaled up (recommendation > requests) or scaled down (recommendation < requests).
+// It also supports >= and <= relations.
+func NewScalingDirectionPodEvictionAdmission() PodEvictionAdmission {
+	return &scalingDirectionPodEvictionAdmission{}
+}
+
+type scalingDirectionPodEvictionAdmission struct {
+	EvictionRequirements map[*apiv1.Pod][]*vpa_types.EvictionRequirement
+}
+
+// Admit denies Pod eviction, if an EvictionRequirement exists for at least one Container which evaluates to 'false'.
+// If no EvictionRequirement exists or no Recommendation is present yet, the Pod is admitted for eviction.
+func (s *scalingDirectionPodEvictionAdmission) Admit(pod *apiv1.Pod, resources *vpa_types.RecommendedPodResources) bool {
+	podEvictionRequirements, found := s.EvictionRequirements[pod]
+	if !found {
+		return true
+	}
+	for _, container := range pod.Spec.Containers {
+		recommendedResources := vpa_utils.GetRecommendationForContainer(container.Name, resources)
+		if recommendedResources == nil {
+			return true
+		}
+		if s.admitContainer(container, recommendedResources, podEvictionRequirements) {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *scalingDirectionPodEvictionAdmission) admitContainer(container apiv1.Container, recommendedResources *vpa_types.RecommendedContainerResources, podEvictionRequirements []*vpa_types.EvictionRequirement) bool {
+	requestedCPU, foundCPURequests := container.Resources.Requests[apiv1.ResourceCPU]
+	recommendedCPU := recommendedResources.Target[apiv1.ResourceCPU]
+	if foundCPURequests && !s.allowEvictionForResourceAndScalingDirection(apiv1.ResourceCPU, requestedCPU.MilliValue(), recommendedCPU.MilliValue(), podEvictionRequirements) {
+		return false
+	}
+
+	requestedMemory, foundMemoryRequests := container.Resources.Requests[apiv1.ResourceMemory]
+	recommendedMemory := recommendedResources.Target[apiv1.ResourceMemory]
+	if foundMemoryRequests && !s.allowEvictionForResourceAndScalingDirection(apiv1.ResourceMemory, requestedMemory.MilliValue(), recommendedMemory.MilliValue(), podEvictionRequirements) {
+		return false
+	}
+	return true
+}
+
+func (s *scalingDirectionPodEvictionAdmission) allowEvictionForResourceAndScalingDirection(resource apiv1.ResourceName, currentRequests int64, recommendation int64, podEvictionRequirements []*vpa_types.EvictionRequirement) bool {
+	for _, evictionRequirement := range podEvictionRequirements {
+		for _, resourceName := range evictionRequirement.Resources {
+			if resourceName == resource {
```

Comment:
if resourceName != resource {
  continue
}

then we have 1 less indent level